### ISSUE TITLE
third game controller button / middle mouse button...

### ISSUE THREAD

**Issue by mithrendal** [dirkwhoffmann/vAmiga#747]
![image](https://user-images.githubusercontent.com/17108995/198883434-6db731db-f666-466c-8dea-a7abb412ebf1.png)

( screenshot AmigaTestKit-1.20/AmigaTestKit.adf )

until now we only got two buttons on a game port ... 
```
      vamiga->controlPort1.mouse.setLeftButton(pressed==1);
      vamiga->controlPort1.mouse.setRightButton(pressed==1);
```

we need one more ... to be able to press the 3rd (CD32) gamepad button or mouse middle button... 

this is a requested feature of eab member @aZtOcKdOg for reference  https://eab.abime.net/showthread.php?p=1571921#post1571921

**Comment by mithrendal** [comment]
found a list of games supporting 2nd button http://eab.abime.net/showthread.php?t=57540

@aZtOcKdOg do you maybe know where the middle mouse button is used ... or if it is used at all on an Amiga ?

**Comment by idrougge** [comment]
The middle button is used to show the volumes list in ASL requesters. It can also be bound using FKey.

**Comment by aZtOcKdOg** [comment]
I will do a few tests today and report back. A few productivity apps can use middle mouse button but I will test if it correlates to the 'third' gamepad btn!

**Comment by mithrendal** [comment]
merged the vAmiga dev branch into vAmigaWeb dev and published to vAmigaWeb UAT (User Acceptance Test) Version at https://vamigaweb.github.io/uat/

![image](https://user-images.githubusercontent.com/17108995/199355934-402b3f7a-b5a0-4a1f-93ba-86fe37864aec.png)


attached the 3 buttons today only to  the physical mouse and physical game controller

game controller buttons are auto mapped by id mod 3, 
e.g. btn1->amiga1, btn2->amiga2, btn3 ->amiga3, btn4->amiga1,...


TODO:
+button2,3 with keyboard joystick emulation
+button2,3 with virtual touch joystick
+type of game controller setting to reduce the supported number of buttons for touch joystick

**Comment by mithrendal** [comment]
![image](https://user-images.githubusercontent.com/17108995/199605106-c3aea26b-1d60-4242-a7a1-71062283bd03.png)


uat test version now supports up to three fire buttons on the virtual touch joystick. 😎

last todo for tomorrow: keyboard joystick emulation support for more fire button 2 and 3 ...

**Comment by mithrendal** [comment]
last todo finished ...  keyboard joystick emulation in uat test version of vAmigaWeb now supports fire button 2 and 3 ...

**Comment by dirkwhoffmann** [comment]
Fixed in v2.2b1
